plugins/git/functions

```diff
@@ -55,16 +55,39 @@ suppress_output() {
   return 0
 }
 
+git_deploy_branch() {
+  declare desc="retrieve the deploy branch for a given application"
+  local cmd="git-hook"
+  local APP="$1"
+
+  local DOKKU_DEPLOY_BRANCH="$(config_get "$APP" DOKKU_DEPLOY_BRANCH || true)"
+  local DOKKU_GLOBAL_DEPLOY_BRANCH="$(config_get --global DOKKU_DEPLOY_BRANCH || true)"
+  if [[ -n "$DOKKU_DEPLOY_BRANCH" ]]; then
+    echo "$DOKKU_DEPLOY_BRANCH"
+  elif [[ -n "$DOKKU_GLOBAL_DEPLOY_BRANCH" ]]; then
+    echo "$DOKKU_GLOBAL_DEPLOY_BRANCH"
```

Comment:
The global bit is inspired by a tweet from some users that didn't want to push `master` because of the connotations of the word. I figure being able to adjust it globally makes most sense.

Also, this makes it so we can specify a `production` box :)